v03_pipeline/lib/tasks/base/base_variant_annotations_table.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+import hail as hl
+import luigi
+
+from v03_pipeline.lib.annotations.fields import get_fields
+from v03_pipeline.lib.model import AnnotationType
+from v03_pipeline.lib.paths import (
+    reference_dataset_collection_path,
+    variant_annotations_table_path,
+)
+from v03_pipeline.lib.tasks.base.base_pipeline_task import BasePipelineTask
+from v03_pipeline.lib.tasks.files import (
+    GCSorLocalFolderTarget,
+    GCSorLocalTarget,
+    HailTableTask,
+)
+
+
+class BaseVariantAnnotationsTableTask(BasePipelineTask):
+    liftover_ref_path = luigi.OptionalParameter(
+        default='gs://hail-common/references/grch38_to_grch37.over.chain.gz',
+        description='Path to GRCh38 to GRCh37 coordinates file',
+    )
+
+    def output(self) -> luigi.Target:
+        return GCSorLocalTarget(
+            variant_annotations_table_path(
+                self.env,
+                self.reference_genome,
+                self.dataset_type,
+            ),
+        )
+
+    def complete(self) -> bool:
+        return GCSorLocalFolderTarget(self.output().path).exists()
+
+    def requires(self) -> list[luigi.Task]:
+        requirements = [
+            HailTableTask(
+                reference_dataset_collection_path(
+                    self.env,
+                    self.reference_genome,
+                    self.dataset_type.base_reference_dataset_collection,
+                ),
+            ),
+        ]
+        for rdc in self.dataset_type.annotatable_reference_dataset_collections(
+            self.env,
+        ):
+            requirements.append(
+                HailTableTask(
+                    reference_dataset_collection_path(
+                        self.env,
+                        self.reference_genome,
+                        rdc,
+                    ),
+                ),
+            )
+        return requirements
+
+    def initialize_table(self) -> hl.Table:
+        if self.dataset_type.base_reference_dataset_collection is None:
+            key_type = self.dataset_type.table_key_type(self.reference_genome)
+            ht = hl.Table.parallelize(
+                [],
+                key_type,
+                key=key_type.fields,
+            )
+        else:
+            ht = hl.read_table(
+                reference_dataset_collection_path(
+                    self.env,
+                    self.reference_genome,
+                    self.dataset_type.base_reference_dataset_collection,
+                ),
+            )
+            ht = ht.annotate(
```

Comment:
I annotated the test reference data with `vep` under the assumption that we'd be doing that... but I think per discussion we can assume it will be unannotated for now and the `vep` annotation will happen here.

Reply:
I thought the agreement was the reference data WILL be annotated with vep, and that we would onyl be annotating the new rows with vep during loading

Reply:
Ah... I thought our conclusion was that the reference data would eventually be annotated with VEP but it was ok for us to leave it un-annotated for now and do the annotation here.  

I actually prefer just doing it now though since the reference data is still ungenerated... so I will proceed with adding the `run_vep` annotation to the `write_combined_ht.py` script.